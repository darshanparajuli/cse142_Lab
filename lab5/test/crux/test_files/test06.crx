// test addition

func addInts(a:int, b:int) : int {
  return a + b;
}

func addFloats(a:float, b:float) : float {
  return a + b;
}

func addBools(a:bool, b:bool) : bool {
  return a + b;
}

func addIntBool(a:int, b:bool) : void {
  return a + b;
}

func addIntFloat(a:int, b:float) : void {
  return a + b;
}

func main() : void {
  ::printFloat(7.0);
}

func addWithError() : void {
  var a : int;
  let a = 1 + 2 + 3 + false + 4 + 5 + 6;
}
