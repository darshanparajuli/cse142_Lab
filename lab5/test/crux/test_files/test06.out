Error type-checking file.
TypeError(12,12)[Cannot add bool with bool.]
TypeError(12,3)[Function addBools returns bool not ErrorType(Cannot add bool with bool.).]
TypeError(16,12)[Cannot add int with bool.]
TypeError(16,3)[Function addIntBool returns void not ErrorType(Cannot add int with bool.).]
TypeError(20,12)[Cannot add int with float.]
TypeError(20,3)[Function addIntFloat returns void not ErrorType(Cannot add int with float.).]
TypeError(29,21)[Cannot add int with bool.]
TypeError(29,29)[Cannot add ErrorType(Cannot add int with bool.) with int.]
TypeError(29,33)[Cannot add ErrorType(Cannot add ErrorType(Cannot add int with bool.) with int.) with int.]
TypeError(29,37)[Cannot add ErrorType(Cannot add ErrorType(Cannot add ErrorType(Cannot add int with bool.) with int.) with int.) with int.]
TypeError(29,3)[Cannot assign ErrorType(Cannot add ErrorType(Cannot add ErrorType(Cannot add ErrorType(Cannot add int with bool.) with int.) with int.) with int.) to Address(int).]

